### PR TITLE
spec: candid type selector

### PR DESCRIPTION
**Pull request by chenyan-dfinity**

### CODE REVIEW

File: spec/Type-selector.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+## Design
+
+Motivated by Cascading Style Sheets ([CSS](https://en.wikipedia.org/wiki/CSS)), we define a configuration language to specify the customization of a Candid interface. The configuration file enables the separation of the Candid interface and its configuration, including layout, binding, validation, etc.
+
+There are two parts in the config language: 1) A way to quickly select particular type nodes in Candid types; 2) Attach semantic meaning to the selected type nodes.
+
+### Type selector
+
+For the first part, we define a type selector `<selector>` similar to CSS selector.
```
Comment (crusso) [inline spec/Type-selector.md:24]:
```suggestion
For the first part, we define a type selector `<selector>` similar to a CSS selector.
```

File: spec/Type-selector.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+```
+<selector> := <scopes> (. <path>)? | <path>
+<path> := <name> (. <name>)*
+<scopes> := <scope> (. <scope>)*
+<scope> := func:<name> | func:init | arg:<num> | ret:<num>
```
Comment (crusso) [inline spec/Type-selector.md:30]: Isn't this missing the `variant` and `record` selectors used in <paths>? As in `tree.variant.branch.record.left` below?

[the remark this answers is not in the document]
Comment (crusso) [reply spec/Type-selector.md:30]: Oh, it's explained further down.

**Review by crusso — CHANGES_REQUESTED** [review]
I'm not saying do this now, but it is possible to give a  semantics to this? Eg mapping a candid AST to  an extended Candid AST with nodes decorated with application specific sets of attributes?

File: spec/Type-selector.md (hunk 2 above)
Comment (crusso) [reply spec/Type-selector.md:30]: Is this resolved? Or did I misunderstand something?

**Review by crusso — APPROVED** [review]
I'll approve but see unresolved issue...

File: spec/Type-selector.md (hunk 2 above)
Comment (chenyan-dfinity) [reply spec/Type-selector.md:30]: Scope only refers to top level functions and arguments. `record` and `variant` is only part of the path. It should become clear when we get the full semantics specified out...

**Review by crusso — APPROVED** [review]

**Pull request merged**